.azure-devops/templates/setup-tox.yml

```diff
@@ -0,0 +1,14 @@
+parameters:
+- name: pythonVersion
+  type: string
+  default: '3.8'
+
+steps:
+  - task: UsePythonVersion@0
+    displayName: Get Python for Python tools.
+    inputs:
+      versionSpec: ${{ parameters.pythonVersion }}
+      addToPath: false
+    name: pyTools
+  - script: $(pyTools.pythonLocation)/bin/pip install --upgrade tox
```

Comment:
similar question/comment as the [script tox invocation](https://github.com/Azure/azure-iot-cli-extension/pull/629/files#r1103260813) , does `python -m pip install ...` not work?

Reply:
Most likely, and I think that's the right solution. This was likely an artifact of some samples I was using when trying to "recreate" a version of [this pipeline](https://github.com/tox-dev/azure-pipelines-template/blob/master/run-tox-env.yml) (from tox) to manage multiple tox jobs using specific python versions.

This allows me to more explicitly say "hey use that very specific version of python we just asked for" (`pyTools`) "and make sure that we're running tox specifically from that interpreter for consistency"

Now that I think about it, once we hard-set a python version using `UsePythonVersion` - it should update path so that `python` points to that location as well, so we may not need to be this explicit.

I'll work on testing this change